erpnext/agriculture/doctype/plant_batch/plant_batch.js

```diff
@@ -47,6 +53,37 @@ frappe.ui.form.on('Plant Batch', {
 			}, __("Create"));
 		}
 
+	},
+	destroy_plant_batch: (frm) => {
+		frappe.prompt([{
+			fieldname: 'destroy_count',
+			label: __('Destroy Count'),
+			fieldtype: 'Int',
+			reqd: 1,
+		},
+		{
+			fieldname: 'reason',
+			label: __('Reason'),
+			fieldtype: 'Data',
+			reqd: 1,
+		}],
+		(data) => {
+			frappe.call({
```

Comment:
Replace this with `frm.call` (check other references in code), and move the `destroy_plant_batch` function into the PlantBatch class

Reply:
cannot move destroy_plant_batch function to PlantBatch class.
as it pass required arguments from .js file

Reply:
You can't use `frm.call` if you don't move the function to the class. `frm.call` does take an `args` parameter that you can still use to pass the arguments.